packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -2449,6 +2449,43 @@ describe('TransactionController', () => {
     });
   });
 
+  describe('clearUnapprovedTxs', () => {
+    it('clears unapproved transactions', async () => {
+      const controller = newController();
+
+      const unapprovedTxId = '1';
+      const confirmedTxId = '2';
+
+      controller.state.transactions.push({
```

Comment:
Should we add a second entry to bulk methods like this to ensure it's iterating correctly?

Reply:
Fixed